### PR TITLE
feat: add support for array-based command setting when running containers via docker run

### LINKED ISSUE #658: Unable to run arbitrary commands in nixpacks-built image
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

I have the following app: https://github.com/dokku/dokku/tree/master/tests/apps/python-flask

It builds successfully with nixpacks:

```shell
nixpacks build --name app:latest .
```

But when I try to run a command, it fails:

```shell
% docker run -it --rm app:latest touch /tmp/derp
touch: missing file operand
Try 'touch --help' for more information.
```

This seems wrong, but I'm not sure what's swallowing up the arguments.

### To reproduce

```shell
git clone https://github.com/dokku/dokku.git
cd dokku/tests/apps/python-flask
nixpacks build --name app:latest .
docker run -it --rm app:latest touch /tmp/derp
```

### Expected behavior

I would expect it to exit 0 and run the command I specified.

### Environment

- nixpacks 0.11.5
- MacOS Monterey 12.6

```
% docker version
Client:
 Cloud integration: v1.0.29
 Version:           20.10.17
 API version:       1.41
 Go version:        go1.17.11
 Git commit:        100c701
 Built:             Mon Jun  6 23:04:45 2022
 OS/Arch:           darwin/arm64
 Context:           default
 Experimental:      true

Server: Docker Desktop 4.12.0 (85629)
 Engine:
  Version:          20.10.17
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.17.11
  Git commit:       a89b842
  Built:            Mon Jun  6 23:01:01 2022
  OS/Arch:          linux/arm64
  Experimental:     false
 containerd:
  Version:          1.6.8
  GitCommit:        9cd3357b7fd7218e4aec3eae239db1f68a5a6ec6
 runc:
  Version:          1.1.4
  GitCommit:        v1.1.4-0-g5fd4c4d
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

### PR DESCRIPTION
**Pull request by josegonzalez**
Previously, all commands had to be quoted, as they were passed as is to an entrypoint that only understood a single argument- /bin/bash -l -c. This caused issues with generic usage of docker run where the command often isn't quoted.

This change allows folks to use either quoted commands (existing) or an array of commands (new). It does so by treating the passed arguments as a single string in all cases via $*.

See https://stackoverflow.com/a/64689634 and https://unix.stackexchange.com/a/129077 for more details on how variable expansion works in shell.

Closes #658

- [ ] Tests are added/updated if needed
- [x] Docs are updated if needed

### CODE REVIEW

**Comment by josegonzalez** [comment]
Not sure what tests need to be changed since a bunch of tests fail for me locally even on master (Running M1 Mac if that matters).

**Comment by josegonzalez** [comment]
How should I add tests? This is what I did locally:

```shell
cargo run -- build --label key=value --name derp examples/python
docker container run --rm -it --entrypoint '' derp cat /usr/local/bin/entrypoint
docker container run --rm -it derp "ls -lah /app && python \"main.py\""
docker container run --rm -it derp ls -lah /app
docker container run --rm -it derp python main.py
```

With the following output:

```
   Compiling nixpacks v1.9.0 (/Users/josediazgonzalez/Dropbox/go/src/github.com/railwayapp/nixpacks)
    Finished dev [unoptimized + debuginfo] target(s) in 2.22s
     Running `target/debug/nixpacks build --label key=value --name derp examples/python`

╔═══════════════════════════════ Nixpacks v1.9.0 ══════════════════════════════╗
║ setup      │ python38, gcc                                                   ║
║──────────────────────────────────────────────────────────────────────────────║
║ install    │ python -m venv --copies /opt/venv && . /opt/venv/bin/activate   ║
║            │ && pip install -r requirements.txt                              ║
║──────────────────────────────────────────────────────────────────────────────║
║ start      │ python main.py                                                  ║
╚══════════════════════════════════════════════════════════════════════════════╝

[+] Building 22.8s (14/14) FINISHED
 => [internal] load build definition from Dockerfile                                                                                                                  0.0s
 => => transferring dockerfile: 1.05kB                                                                                                                                0.0s
 => [internal] load .dockerignore                                                                                                                                     0.0s
 => => transferring context: 2B                                                                                                                                       0.0s
 => [internal] load metadata for ghcr.io/railwayapp/nixpacks:ubuntu-1684957838                                                                                        0.2s
 => CACHED [stage-0 1/9] FROM ghcr.io/railwayapp/nixpacks:ubuntu-1684957838@sha256:eb26a5ad60faad269d01ac896a4eefb5e7987040d39549dbe5d8cdfd1a830b75                   0.0s
 => [internal] load build context                                                                                                                                     0.0s
 => => transferring context: 2.38kB                                                                                                                                   0.0s
 => [stage-0 2/9] RUN printf '#!/usr/bin/env bash\nexec bash -l -c -- "$*"\n' > /usr/local/bin/entrypoint && chmod +x /usr/local/bin/entrypoint                       0.3s
 => [stage-0 3/9] WORKDIR /app/                                                                                                                                       0.0s
 => [stage-0 4/9] COPY .nixpacks/nixpkgs-293a28df6d7ff3dec1e61e37cc4ee6e6c0fb0847.nix .nixpacks/nixpkgs-293a28df6d7ff3dec1e61e37cc4ee6e6c0fb0847.nix                  0.0s
 => [stage-0 5/9] RUN nix-env -if .nixpacks/nixpkgs-293a28df6d7ff3dec1e61e37cc4ee6e6c0fb0847.nix && nix-collect-garbage -d                                           17.8s
 => [stage-0 6/9] COPY . /app/.                                                                                                                                       0.1s
 => [stage-0 7/9] RUN --mount=type=cache,id=ApobWLr7gNk-/root/cache/pip,target=/root/.cache/pip python -m venv --copies /opt/venv && . /opt/venv/bin/activate && pip  3.1s
 => [stage-0 8/9] RUN printf '\nPATH=/opt/venv/bin:$PATH' >> /root/.profile                                                                                           0.5s
 => [stage-0 9/9] COPY . /app                                                                                                                                         0.1s
 => exporting to image                                                                                                                                                0.9s
 => => exporting layers                                                                                                                                               0.9s
 => => writing image sha256:45bd914a9cbffe5c82d9da4e6290e8db7a3832841ddc164a7f8c7c8d7334a5c0                                                                          0.0s
 => => naming to docker.io/library/derp                                                                                                                               0.0s
=== Successfully Built! ===

Run:
  docker run -it derp
#!/usr/bin/env bash
exec bash -l -c -- "$*"
total 8.0K
drwxr-xr-x 1 root root  64 May 26 03:48 .
drwxr-xr-x 1 root root   0 May 26 03:48 ..
drwxr-xr-x 1 root root 124 May 26 03:48 .nixpacks
-rw-r--r-- 1 root root  27 May 26 03:48 main.py
-rw-r--r-- 1 root root 112 May 26 03:48 requirements.txt
Hello from Python
total 8.0K
drwxr-xr-x 1 root root  64 May 26 03:48 .
drwxr-xr-x 1 root root   0 May 26 03:48 ..
drwxr-xr-x 1 root root 124 May 26 03:48 .nixpacks
-rw-r--r-- 1 root root  27 May 26 03:48 main.py
-rw-r--r-- 1 root root 112 May 26 03:48 requirements.txt
Hello from Python
```

**Comment by coffee-cup** [comment]
This is neat! I will do some testing with how we use Railway internally and hope to get this merged this week

**Comment by josegonzalez** [comment]
@coffee-cup long shot, but anything I can do to help out with testing?

**Comment by coffee-cup** [comment]
> @coffee-cup long shot, but anything I can do to help out with testing?

Hey @josegonzalez, sorry I haven't had a solid chance to dig into this yet. I would still like to get this merged and will hopefully have time this week. Normally we would get this merged much quicker, but this change effects all apps deployed with Nixpacks so I just want to give it the attention it deserves.

**Comment by josegonzalez** [comment]
No worries! This one is a bit selfish of a PR for me, so let me know if I can help in any way :)

**Comment by josegonzalez** [comment]
If I re-issue this PR, is it likely to be reviewed/merged? If there isn't any interest, I'll just hold off on doing so.

File: src/nixpacks/builder/docker/dockerfile_generation.rs [hunk 1]
```diff
@@ -23,6 +23,11 @@ use std::{
 const NIXPACKS_OUTPUT_DIR: &str = ".nixpacks";
 pub const APP_DIR: &str = "/app/";
 
+const CREATE_ENTRYPOINT: &str = "
+RUN printf '#!/usr/bin/env bash\\nexec bash -l -c -- \"$*\"\\n' > /usr/local/bin/entrypoint && \
+    chmod +x /usr/local/bin/entrypoint
+";
+
```
Comment (josegonzalez) [inline src/nixpacks/builder/docker/dockerfile_generation.rs:30]: If someone wants to revive this PR, change the run command like so:

```
RUN printf '#!/usr/bin/env bash\nexec bash -l -c -- \"$*\"\n' > /usr/local/bin/entrypoint && \
    chmod +x /usr/local/bin/entrypoint
```

**Comment by coffee-cup** [comment]
There is still interest in this. But will likely be added when we cut the next major release of Nixpacks

**Comment by josegonzalez** [comment]
Feel free to take the code - the github-actions automation on stale PRs is annoying to deal with since it'll just close the PR unless I rebase every few weeks.

**Pull request closed without merging**